### ISSUE TITLE
[BUG] Version reports 1.0.21, but 1.0.18 is checked out.

### ISSUE THREAD

**Issue by borntohonk** [linuxserver/docker-medusa#55]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Branch: [master](https://github.com/pymedusa/Medusa/tree/master)
Commit: [b8bec48d5036a4d9084d762e2d2ffca9c711f2f3](https://github.com/pymedusa/Medusa/commit/b8bec48d5036a4d9084d762e2d2ffca9c711f2f3)
Version: [1.0.21](https://github.com/pymedusa/Medusa/releases/tag/v1.0.21)

https://github.com/pymedusa/Medusa/releases/tag/v1.0.18 = https://github.com/pymedusa/Medusa/commit/b8bec48d5036a4d9084d762e2d2ffca9c711f2f3

### Expected Behavior

expected behavior is for https://github.com/pymedusa/Medusa/commit/34a67cf159acab69017dd5b95e0233ecde0de28a to be checked out, not https://github.com/pymedusa/Medusa/commit/b8bec48d5036a4d9084d762e2d2ffca9c711f2f3

also docker container flag seems to be unset, and there's no git within the container.

### Steps To Reproduce

1. run the container and notice version flag has been tampered with.
2. debug the container and find no git binary.
3. apk add git
4. cd /app/medusa
5. git status
6. notice that it's not initialized, and have been tampered with, instead of being checked out and having the docker container flag which medusa supports not set.

### Environment

```markdown
- OS: Truenas Scale
- How docker service was installed: integrated service through kubernetes/HELM




```
![image](https://github.com/linuxserver/docker-medusa/assets/6264306/f25d5a8b-4fb4-4789-aa00-9b2c1838555c)

### CPU architecture

x86-64

### Docker creation

```bash
https://github.com/truecharts/charts/tree/master/charts/stable/medusa
```

### Container logs

```bash
the logs themselves are filled with medusa related work / downloads / processing and are worthless to this bug report.
```

**Comment by Roxedus** [comment]
Where is the tamper conclusion coming from? both medusa/common.py and CHANGELOG.md claims 1.0.21.

We do indeed not use git to grab the version, and rather the source-code tarball.

**Comment by borntohonk** [comment]
It's an assumption based on incorrect commit checkout being set,
`MEDUSA_COMMIT_HASH` should be set to prevent this behavior. 
![image](https://github.com/linuxserver/docker-medusa/assets/6264306/fb06a4dc-105a-4e65-bf10-e2c729234ddd)


ARG GIT_COMMIT
ENV MEDUSA_COMMIT_HASH $GIT_COMMIT

(the official pymedusa docker container sets this through workflows)

```
        build-args: |
          GIT_COMMIT=${{ github.sha }}
          GIT_BRANCH=${{ github.ref_name }}
````

**Comment by aptalca** [comment]
We don't `check out` anything, we don't use `git`. 

We just grab the github release tarball: https://github.com/linuxserver/docker-medusa/blob/master/Dockerfile#L30-L32

**Comment by Roxedus** [comment]
We do not set/use/need that environment variable at all

**Comment by borntohonk** [comment]
> We do not set/use/need that environment variable at all

And that leads to this behavior where the container chooses to identify as an older cached version

**Comment by aptalca** [comment]
Honestly it's quite silly to rely on git commits for version checks on stable releases. It's appropriate for development releases and dev purposes but not stable releases for public consumption. I realize various FOSS projects used to do that back in the day, including for webgui based updates, but most have moved on to proper versioning for stable releases.

With that said, we don't support internal update mechanisms via the gui. We used to go out of our way to try and disable those when it used to be common. These days it's rare so we no longer bother.

In short, if you're using our docker image, don't use web gui update mechanisms. The update instructions as well as instructions for checking the app version within containers are both described in the readme:
- https://github.com/linuxserver/docker-medusa?tab=readme-ov-file#support-info
- https://github.com/linuxserver/docker-medusa?tab=readme-ov-file#updating-info

With regards to the `tampering`  concerns, we take those very seriously but would need a lot more info than what is currently provided to test and confirm.

**Comment by Roxedus** [comment]
That test is at [best flakey](github.com/pymedusa/Medusa/blob/34a67cf159acab69017dd5b95e0233ecde0de28a/medusa/updater/version_checker.py#L340-L363). 

There is no commit tied to a release on github(it can be tied to a git tag, which does have a commit associated).
The fallback is to look for a `.dockerenv` file, which does indeed exist in a docker environment.
![image](https://github.com/linuxserver/docker-medusa/assets/7110194/c18c1d16-0578-422b-9255-f4cd175c3693)

**Comment by borntohonk** [comment]
The issue here is i use persistent storage mounted (medusa configuration) and that has indexed commit derived from prior version set and cached that.

I am using the container :latest container 1:1. One needs to set `MEDUSA_COMMIT_HASH` otherwise the pymedusa container will insist there are X amount of commits and in need of an update.

**Comment by aptalca** [comment]
Isn't the indexed commit stored in the config file? Can't you just remove it as it never should have been set in our container to begin with? It's a side effect of migrating from either a different image or bare metal to our container and should be part of the migration steps.

**Comment by borntohonk** [comment]
You're basically suggesting I go do surgery to my config file every iteration of the container that edits common.py to change the version to a higher increment, instead of just setting the enviroment variable to prevent this behavior from happening?
edit:
just a note: it's more than "just a config file"; it's a whole system of database entries, to label it as "just a config file" is underplaying what needs to be done.

**Comment by Roxedus** [comment]
We haven't exposed the commit SHA for years in Medusa. So how it even knew which to put there is a mystery. 

The time for that 43rd commit in Medusa tracks with when Truecharts decided to be less shitty by rebranding containers, who knows what they did to the images, maybe they set the env once.

**Comment by borntohonk** [comment]
a live test of my database is that it bricked editing a config file to set the commit hash, and reapplying backups don't work. It's a fickle webserver from a stagnant codebase. The container itself runs.

**Comment by aptalca** [comment]
I'm in a meeting and I haven't used or looked into Medusa in a loong time so I'm going by what you said.

> The issue here is i use persistent storage mounted (medusa configuration) and that has indexed commit derived from prior version set and cached that.

I'm assuming you're talking about a migration related issue. Because when I put up a container just now, an update check tells me I'm already on latest and it also tells me the docker message Rox posted above.

Also like Rox said, we are not the ones publishing helm charts. Who knows what they did in the past.

**Comment by borntohonk** [comment]
> I'm in a meeting and I haven't used or looked into Medusa in a loong time so I'm going by what you said.
> 
> > The issue here is i use persistent storage mounted (medusa configuration) and that has indexed commit derived from prior version set and cached that.
> 
> I'm assuming you're talking about a migration related issue. Because when I put up a container just now, an update check tells me I'm already on latest and it also tells me the docker message Rox posted above.
> 
> Also like Rox said, we are not the ones publishing helm charts. Who knows what they did in the past.

oh well, my database is already lost so guess i have to rebuild new anyway and assume this is a former truecharts thing, and just reopen a new issue if it turns out not to be.

**Comment by aptalca** [comment]
For future reference (while I have the container up)

Here's the help and info page that shows the commit as `unknown` and the version correctly as `1.0.21`. Also shows `Runs in Docker: Yes`
![Screenshot 2024-04-16 104730](https://github.com/linuxserver/docker-medusa/assets/541623/e4453783-f87f-4e6b-b3b9-103534f4e60c)

Also, `/config/config.ini` lists:
```
[General]
git_token = ""
git_reset = 1
git_reset_branches = ,
branch = master
git_remote = origin
git_remote_url = https://github.com/pymedusa/Medusa
cur_commit_hash = ""
cur_commit_branch = master
config_version = 12
```

What I was trying to tell you earlier is to reset that option `cur_commit_hash = ""` in there, which should never be set in our container. And if set due to migration, would need to be reset, otherwise you'll get incorrect update info in the webgui.
